### PR TITLE
(BKR-466) added beaker-library instructions

### PR DESCRIPTION
**Pull request by kevpl**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,132 @@
 # beaker-template
-Let's try and construct a Beaker DSL extension library!
+
+Let's construct a Beaker DSL extension library!
+
+# Why Would We Do Such a Thing?
+
+There are two reasons to create a Beaker library:
+
+1. To pull functionality out of Beaker core to be maintained/improved separately (mostly QE tasks)
```
Comment (colinPL) [inline README.md:9]: The text in parens can be removed.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,132 @@
...
+2. To provide additional functionality to their Beaker testing (biggest Beaker user use case)
```
Comment (colinPL) [inline README.md]: It is not clear what this point means. "To provide additional functionality to beaker" is what it is saying?

File: README.md (hunk 1 above)
Comment (kevpl) [inline README.md:9]: The text in parens is there to give people an idea of who this use case applies to.

I'd rather include too much info than too little.  Is there some reason it should be cut?

File: README.md (hunk 2 above)
Comment (kevpl) [inline README.md]: Yeah, that's the use case: when you want to do something beaker can't, so you're providing additional functionality to beaker.  I'm confused about which part is confusing.

Is it just the wording?  Would something like "to provide functionality beaker doesn't have" make more sense to you?

Comment (anodelman) [inline README.md]: I would go with "To provide additional methods to the Beaker DSL, extending Beaker functionality."

Comment (kevpl) [inline README.md]: @colinPL does that sound good for you?

Comment (colinPL) [inline README.md]: That sounds good. And remove the text in parens :smiley_cat:

Comment (kevpl) [inline README.md]: why?

Comment (colinPL) [inline README.md]: The paren text is fine and is not blocking. I do think the rewording would be beneficial.

**Comment by kevpl** [comment]
@anodelman are we good for final review?

**Comment by kevpl** [comment]
@anodelman I didn't setup this repo with automatic PR Jenkins runs.  There is a test running Jenkins job for this project.  Maybe we should set that up & do a quick run on this?

**Comment by anodelman** [comment]
Well, it's a template, so I don't think that there is much that we can run in way of testing. :)

**Pull request merged**